tools/bqtag/bqtag/bqtag.py

```diff
@@ -0,0 +1,800 @@
+            LOGGER.error("Could not download source table schema: %s", str(e))
+            LOGGER.error(traceback.format_exc())
+            return ""
+
+        f = io.StringIO("")
+        self.bq.schema_to_json(table.schema, f)
+        table_schema = f.getvalue()
+
+        LOGGER.info("Table schema downloaded. Started View Creation.")
+
+        # Create a map of tags to columms
+        tag_column_map = self._create_tag_column_map(schema=table_schema)
+
+        # Find columns having tag in tags
+        columns = []
+
+        for tag in tags:
+            if tag in tag_column_map:
+                columns += tag_column_map[tag]
+
+        # Create column tree and generate query for the view
+        column_tree = ColumnTree()
+
+        for column in columns:
+            column_tree.add_node(column)
+
+        query_columns = column_tree.generate_query()
+
+        if query_columns.strip() == "":
+            LOGGER.info("No Columns in Query. View not created.")
+            return ""
+
+        query = "SELECT {q} FROM `{p}.{d}.{t}`".format(q=query_columns,
+                                                       p = self.bq_project,
+                                                       d = self.dataset,
+                                                       t = table_name
+                                                    )
+
+        LOGGER.debug("Query Created: %s", query)
+
+        # create view
+        job_config = bigquery.QueryJobConfig(destination=".".join(
+                                                    [
+                                                        self.bq_project,
+                                                        self.dataset,
+                                                        view_name
+                                                    ]
+                                                )
+                                            )
+
+        try:
+            query_job = self.bq.query(query, job_config=job_config)
+            query_job.result()  # Wait for the job to complete.
+        except (MethodNotImplemented, NotFound) as e:
+            LOGGER.error("Could not create view: %s", str(e))
+            LOGGER.error(traceback.format_exc())
+            return ""
+
+        LOGGER.info(
+                        "Authorised view created: %s.%s.%s",
+                        self.bq_project,
+                        self.dataset,
+                        view_name
+                   )
+
+        return query
+
+
+    # Internal Function to detemine the tagged schema
+    def _process_schema(self, table_schema: str, table_tag_map: dict) -> None:
+        """
+        Process the untagged schema and convert it into tagged one by
+        adding tags to relevant columns using table_tag_map
+
+        :param table_schema: Untagged schema of the table
+        :param table_tag_map: Mapping of tags to columns
+        :return None
+        """
+
```

Comment:
Remove extra blank line (only class docstrings are followed by a blank line, as per PEP 257)

Reply:
Changed to

```
        """
        Process the untagged schema and convert it into tagged one by.
        adding tags to relevant columns using table_tag_map

        :param table_schema: Untagged schema of the table
        :param table_tag_map: Mapping of tags to columns
        :return: None
        """
        json_schema = json.loads(table_schema)
```

Reply:
Thank you for providing the details as well.